src/Compilers/CSharp/Portable/Syntax/CSharpSyntaxTree.LazySyntaxTree.cs

```diff
@@ -0,0 +1,188 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Text;
+using System.Threading;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    public partial class CSharpSyntaxTree
+    {
+        private sealed class LazySyntaxTree : CSharpSyntaxTree
+        {
+            private readonly SourceText _text;
+            private readonly CSharpParseOptions _options;
+            private readonly string _path;
+            private readonly ImmutableDictionary<string, ReportDiagnostic> _diagnosticOptions;
+            private CSharpSyntaxNode? _lazyRoot;
+
+            internal LazySyntaxTree(
+                SourceText text,
+                CSharpParseOptions options,
+                string path,
+                ImmutableDictionary<string, ReportDiagnostic>? diagnosticOptions)
+            {
+                Debug.Assert(options != null);
+
+                _text = text;
+                _options = options;
+                _path = path ?? string.Empty;
```

Comment:
>path [](start = 24, length = 4)

`path` is declared as not `null`.

Reply:
So for regular trees this is what we're doing as well, which seems odd. We always silently converted a null going in to an empty string. When we annotated the methods we then kept them as non-nullable to discourage people from passing null (since it's probably not entirely what they wanted....) but it was still allowed and silently changed.

This is matching the other types. I _can_ change this, but it'll be inconsistent unless I change the other types as well. Thoughts?

Reply:
Let's keep it as is, and consistent. Thanks for the explanation.

---
In reply to: [601792055](https://github.com/dotnet/roslyn/pull/50490#discussion_r601792055) [](ancestors = 601792055)